packages/panna-sdk/src/core/util/api-service.ts

```diff
@@ -227,6 +236,80 @@ export class PannaApiService {
       throw error;
     }
   }
+
+  /**
+   * Retrieves the status of an onramp.money session
+   *
+   * This function requests the Panna API to get the current status of a fiat-to-crypto
+   * onramp session. The status includes transaction details, amounts, and any error messages.
+   *
+   * @param params - Parameters for retrieving the session status
+   * @param params.sessionId - The onramp.money session identifier
+   * @param params.authToken - JWT token for authentication
+   * @returns Promise resolving to the session status with all transaction details
+   * @throws Error if the session ID is invalid or network request fails
+   */
+  public async getSessionStatus(
+    params: GetSessionStatusParams
+  ): Promise<SessionStatusResult> {
+    const { baseUrl, isMockMode } = this.config;
+    const { sessionId, authToken } = params;
+
+    if (!sessionId) {
+      throw new Error('Session ID is required');
+    }
+
+    if (isMockMode) {
+      await new Promise((resolve) => setTimeout(resolve, 500));
+
+      if (sessionId.includes(OnrampMoneySessionStatusEnum.Created)) {
+        return mockSessionStatusCreated;
+      } else if (sessionId.includes(OnrampMoneySessionStatusEnum.Pending)) {
+        return mockSessionStatusPending;
+      } else {
+        return mockSessionStatusCompleted;
+      }
+    }
+
+    if (!authToken) {
+      throw new Error('Auth token is required to fetch session status');
+    }
```

Comment:
```suggestion
    if (!sessionId) {
      throw new Error('Session ID is required');
    }

    if (!authToken) {
      throw new Error('Auth token is required to fetch session status');
    }

    if (isMockMode) {
      await new Promise((resolve) => setTimeout(resolve, 500));

      if (sessionId.includes(OnrampMoneySessionStatusEnum.Created)) {
        return mockSessionStatusCreated;
      } else if (sessionId.includes(OnrampMoneySessionStatusEnum.Pending)) {
        return mockSessionStatusPending;
      } else {
        return mockSessionStatusCompleted;
      }
    }
```

Reply:
I initially did it like this then switched it because mock mode tests would fail if they don't get passed the auth token